src/Codegen/Builders/ObjectBuilder.hack

```diff
@@ -19,6 +43,20 @@ final class ObjectBuilder<TField as IFieldBuilder> extends CompositeBuilder<TFie
             $type_info,
             $type_alias->getName(),
             Vec\map_with_key($ts['fields'], ($name, $ts) ==> new ShapeFieldBuilder($name, $ts)),
+            $hack_class_to_graphql_interface,
         );
     }
+
+    private function generateInterfacesConstant(HackCodegenFactory $cg): CodegenClassConstant {
+        $interfaces = dict[];
+        foreach ($this->hack_class_to_graphql_interface as $hack_class => $graphql_type) {
+            if (\is_subclass_of($this->hack_type, $hack_class)) {
+                $interfaces[$graphql_type] = Str\format('%s::class', $hack_class);
```

Comment:
I think we want the generated GraphQL class name here (which AFAIK is always equivalent to `$graphql_type` right now), not the original Hack class, or not?

```suggestion
                $interfaces[$graphql_type] = Str\format('%s::class', $graphql_type);
```

Reply:
Good catch, updated with a test too to catch these being named differently